api/src/main/java/org/openmrs/module/fhir2/providers/r3/EncounterFhirResourceProvider.java

```diff
@@ -135,6 +136,7 @@ public IBundleProvider searchEncounter(@OptionalParam(name = Encounter.SP_DATE)
 	        @OptionalParam(name = Encounter.SP_PATIENT, chainWhitelist = { "", Patient.SP_IDENTIFIER, Patient.SP_GIVEN,
 	                Patient.SP_FAMILY, Patient.SP_NAME }, targetTypes = Patient.class) ReferenceAndListParam patientParam,
 	        @OptionalParam(name = Encounter.SP_RES_ID) TokenAndListParam id,
+	        @OptionalParam(name = Encounter.SP_TYPE) CodeableConcept type,
```

Comment:
Hope this is the right track. 
Should the type be `org.hl7.fhir.r4.model.CodeableConcept` or `ReferenceAndListParam`
cc @ibacher @mozzy11 @pmanko

Reply:
It should actually be `TokenAndListParam` like `id` above. FHIR's search API has [different types](https://www.hl7.org/fhir/search.html#ptypes) than FHIR's resource representations, but in general if the datatype is a `CodeableConcept` the search parameter will be a `token`, which is the type given for [the `type` search parameter](https://www.hl7.org/fhir/encounter.html#search)